src/Schema/ViewHelperFinder.php

```diff
@@ -77,6 +90,8 @@ private function findViewHelperFilesInPath(string $namespace, string $path): arr
                 $viewHelpers[] = $this->viewHelperMetadataFactory->createFromViewhelperClass($className);
             } catch (\InvalidArgumentException) {
                 // Just ignore this class
+            } catch (\Throwable $t) {
+                $this->lastErrors[] = $t->getMessage();
```

Comment:
You mean something like: "Catch all exceptions to avoid aborting the whole schema generation" ?

Reply:
Catching \Throwable is usually a sign for bad code, or something else is wrong: Errors and Exceptions are different things, and Errors *should* stop processing and shouldn't be caught. Having a catch for a \Throwable, thus should have a comment on why it's needed here, which problem is worked around, and what would happen if it would be left out.

In this case, catching \Throwable mitigates a technical debt: The VH's are not fully and properly set up, so one may run into edge cases when the argument initilization has funny dependencies, which is seldom, though. The partial init is of course not cool, but can't be fully resolved currently in this area since we'd then for instance need all dependencies and inject them and eventually have a running system, and well, not feasible for VH docs generation. Solving this probably involves handling arguments differently - by turning them from code into declaration ... In the end, the \throwable here now shows there are technical debts. And that's what should be documented.

Reply:
Thanks for the explanation. Done.